### DISCUSSION TITLE
OnlyOffice Delayed Start Issue

### DISCUSSION THREAD

**Discussion by Bruceumba** [nextcloud/all-in-one#5693] [Questions] [answered]
I get an error most days when the backup is run and the containers all start booting up:
```
ServerException
Server error: `GET https://cloud.kf7k.com/onlyoffice/healthcheck` resulted in a `502 Bad Gateway` response: <html> <head><title>502 Bad Gateway</title></head> <body> <center><h1>502 Bad Gateway</h1></center> <hr><center>ngin (truncated...)
healthcheckRequest on check error
```
I think that OnlyOffice takes so long to start up that the Nextcloud container runs a check to see if the OnlyOffice container is ready before it's ready. When it fails the check it silently disables the OnlyOffice apps until I go into Administrative Settings/ONLYOFFICE and hit the Save button to rerun the check. Then OnlyOffice works fine.

How can I fix this behavior? Or is it best fixed in programming? Can I add a startup delay to the Nextcloud container? I don't know Docker well enough to add such a setting myself.

**Comment by szaimen** [comment, marked as answer]
Hi, there is a startup delay before the --check command is run:
https://github.com/nextcloud/all-in-one/blob/a5829679deac6b270a66bb427d1acb7bb4e6939d/Containers/nextcloud/run-exec-commands.sh#L3

The slepp only starts after the nextcloud container was completely started, so usually this should be enough time especially since the nextcloud container waits for onlyoffice to start: https://github.com/nextcloud/all-in-one/blob/a5829679deac6b270a66bb427d1acb7bb4e6939d/Containers/nextcloud/entrypoint.sh#L675
